### DISCUSSION TITLE
ASTER

### DISCUSSION THREAD

**Discussion by RichardScottOZ** [gjoseph92/stackstac#56] [Ideas]
Back to having a look at NASA stac
https://github.com/RichardScottOZ/Pangeo-Experiments/blob/main/STAC-Catalogue-NASA-LPDAAC-ECS-AST.ipynb

e.g.
``` python
stac_items = Search(url='https://cmr.earthdata.nasa.gov/stac/LPDAAC_ECS',
                 collections=['AST_L1T.v003'], 
                 bbox = '-74.5,40.2128,-73.5,41.2128' ,                    
                 datetime='2020-04-23/2021-04-23', 
                ).items()

stac_items[1].assets

{'04/AST_L1T_00305042020155106_20200505100108_3340': {'title': 'AST_L1T_00305042020155106_20200505100108_3340.hdf. MimeType: application/x-hdfeos',
  'href': 'https://e4ftl01.cr.usgs.gov//ASTER_L1T/ASTT/AST_L1T.003/2020.05.04/AST_L1T_00305042020155106_20200505100108_3340.hdf',
  'type': 'application/x-hdfeos'},
 '04/AST_L1T_00305042020155106_20200505100108_3340_T': {'title': 'AST_L1T_00305042020155106_20200505100108_3340_T.tif. MimeType: application/x-geotiff',
  'href': 'https://e4ftl01.cr.usgs.gov//ASTER_L1T/ASTT/AST_L1T.003/2020.05.04/AST_L1T_00305042020155106_20200505100108_3340_T.tif',
  'type': 'application/x-geotiff'},
 '04/AST_L1T_00305042020155106_20200505100108_3340_V': {'title': 'AST_L1T_00305042020155106_20200505100108_3340_V.tif. MimeType: application/x-geotiff',
  'href': 'https://e4ftl01.cr.usgs.gov//ASTER_L1T/ASTT/AST_L1T.003/2020.05.04/AST_L1T_00305042020155106_20200505100108_3340_V.tif',
  'type': 'application/x-geotiff'},
 'browse': {'title': 'This Browse file may be downloaded directly from this link (BROWSE)',
  'href': 'https://e4ftl01.cr.usgs.gov//WORKING/BRWS/Browse.001/2020.05.05/AST_L1T_00305042020155106_20200505100108_3340_BR.2.VNIR.jpg',
  'type': 'image/jpeg'},
 'metadata': {'href': 'https://cmr.earthdata.nasa.gov/search/concepts/G1735268278-LPDAAC_ECS.xml',
  'type': 'application/xml'}}
```

unique asset names, but invidiual tifs - so hack that like this?

``` python
import copy
stac_new = copy.deepcopy(stac_items)

for index, i in enumerate(stac_items):
    #print(i.assets)
    for k in i.assets:
        if '_T' in k:
            stac_new[index].assets['B01'] = stac_items[index].assets[k]
        if '_V' in k:
            stac_new[index].assets['B02'] = stac_items[index].assets[k]

stack = stackstac.stack(stac_new, epsg=6933, resolution=90, resampling=1, assets=['B01'])
#stack = stackstac.stack(stac_items, epsg=6933, resolution=30, resampling=1)

stack
```
![image](https://user-images.githubusercontent.com/72196131/119211241-1f00de00-baf0-11eb-8e39-5451d38f8d11.png)

**Comment by gjoseph92** [comment]
What about https://planetarycomputer.microsoft.com/dataset/aster-l1t? Might be a bit easier to work with.

**Reply by RichardScottOZ** [reply]
Only if you have access!  I filled in the form a few weeks ago and got a reply yesterday?  Saying you are on the list, we will get to applications eventually, basically.

No idea if they will be like google though and given it is an enviro thing not allow mining companies to use it.  Or if it will have our needed commercial type scale, either to start with?

All of google of course being run on metal, so somewhat ironic. :)

**Reply by gjoseph92** [reply]
You don't actually have to be registered to use the data (I didn't realize this at first either):

> Anyone can get a token from our Data Authentication API - you don’t have to be a registered Planetary Computer user to get an access token. If you are a registered user, and/or if you are requesting data inside the West Europe Azure Region, the rate limiting and token expiry will be more favorable to doing large-scale work with our data.
https://planetarycomputer.microsoft.com/docs/concepts/sas/

**Comment by RichardScottOZ** [comment]
Thanks Gabe - sounds like a weekend test to try.
